src/Fixer/Phpdoc/PhpdocTrimAfterDescriptionFixer.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\DocBlock\Line;
+use PhpCsFixer\DocBlock\ShortDescription;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Nobu Funaki <nobu.funaki@gmail.com>
+ */
+final class PhpdocTrimAfterDescriptionFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Removes extra blank lines after Summary and after Description in phpdocs.',
+            [
+                new CodeSample(
+                    '<?php
+/**
+ * Summary.
+ *
+ *
+ * Description.
+ *
+ *
+ * @param string $foo
+ */
+function fnc($foo) {}
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $summaryEnd = (new ShortDescription($doc))->getEnd();
+
+            if (null === $summaryEnd) {
+                continue;
+            }
+
+            $this->fixSummary($doc, $summaryEnd);
+            $this->fixDescription($doc, $summaryEnd);
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $doc->getContent()]);
```

Comment:
not sure, but shouldn't we check if the content remained the same so no new token has to be set? (cc @keradus )

Reply:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/src/Tokenizer/Tokens.php#L314